src/panels/KaitoTestPanel.ts

```diff
@@ -0,0 +1,146 @@
+        readonly kubectl: k8s.APIAvailable<k8s.KubectlV1>,
+        readonly kubeConfigFilePath: string,
+        readonly sessionProvider: ReadyAzureSessionProvider,
+        readonly modelName: string,
+    ) {
+        this.clusterName = clusterName;
+        this.subscriptionId = subscriptionId;
+        this.resourceGroupName = resourceGroupName;
+        this.armId = armId;
+        this.kubectl = kubectl;
+        this.kubeConfigFilePath = kubeConfigFilePath;
+        this.sessionProvider = sessionProvider;
+        this.modelName = modelName;
+    }
+    getTitle(): string {
+        return `Test KAITO Model`;
+    }
+
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName,
+            modelName: this.modelName,
+            output: "",
+        };
+    }
+    getTelemetryDefinition(): TelemetryDefinition<"kaitoTest"> {
+        return {
+            queryRequest: true,
+        };
+    }
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        void webview;
+        return {
+            queryRequest: (params) => {
+                this.handleQueryRequest(
+                    params.prompt,
+                    params.temperature,
+                    params.topP,
+                    params.topK,
+                    params.repetitionPenalty,
+                    params.maxLength,
+                    webview,
+                );
+            },
+        };
+    }
+
+    nullIsFalse(value: boolean | null): boolean {
+        return value ?? false;
+    }
+    // Tracks if query is currently beign sent. If so, prevents user from sending another query
+    private sendingQuery: boolean = false;
+
+    // Sanitizing the input string
+    private escapeSpecialChars(input: string) {
+        return input
+            .replace(/\\/g, "\\\\") // Escape backslashes
+            .replace(/"/g, '\\"') // Escape double quotes
+            .replace(/'/g, "") // Remove single quotes
+            .replace(/\n/g, "\\n") // Escape newlines
+            .replace(/\r/g, "\\r") // Escape carriage returns
+            .replace(/\t/g, "\\t") // Escape tabs
+            .replace(/\f/g, "\\f") // Escape form feeds
+            .replace(/`/g, "") // Remove backticks
+            .replace(/\0/g, "\\0"); // Escape null characters
+    }
+
+    // Sends a request to the inference server to generate a response to the given prompt
+    private async handleQueryRequest(
+        prompt: string,
+        temperature: number,
+        topP: number,
+        topK: number,
+        repetitionPenalty: number,
+        maxLength: number,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        // prevents user from re-submitting while a query is in progress
+        if (this.sendingQuery) {
+            return;
+        }
+        await longRunning(`Sending query...`, async () => {
+            this.sendingQuery = true;
+            const localPort = await getPort();
```

Comment:
getPort is library that gives you random port? can you please leave comment here for other folks to know.